src/components/ApplicationGroup/Details/EnvironmentOverview/RestartWorkloadModal.tsx

```diff
@@ -0,0 +1,663 @@
+        if (expandedAppIds.includes(appId)) {
+            setExpandedAppIds(expandedAppIds.filter((id) => id !== appId))
+        } else {
+            setExpandedAppIds([...expandedAppIds, appId])
+        }
+    }
+
+    const renderHeaderSection = (): JSX.Element => {
+        return (
+            <div className="flex dc__content-space dc__border-bottom pt-12 pr-20 pb-12 pl-20">
+                <div className="fs-16 fw-6 lh-1-5">{` Restart workloads on '${envName}'`}</div>
+                <Close className="icon-dim-24 cursor" onClick={closeDrawer} />
+            </div>
+        )
+    }
+
+    const handleWorkloadSelection = (
+        appId: number,
+        _kindName: string,
+        key: APP_DETAILS_TEXT.APP_NAME | APP_DETAILS_TEXT.KIND_NAME | APP_DETAILS_TEXT.ALL,
+    ) => {
+        const _bulkRotatePodsMap = { ...bulkRotatePodsMap }
+
+        if (key === APP_DETAILS_TEXT.APP_NAME && _bulkRotatePodsMap[appId].appName) {
+            _bulkRotatePodsMap[appId].isChecked = _bulkRotatePodsMap[appId].value !== CHECKBOX_VALUE.CHECKED
+            _bulkRotatePodsMap[appId].value = _bulkRotatePodsMap[appId].isChecked ? CHECKBOX_VALUE.CHECKED : null
+
+            // handling app level value for checkbox
+            Object.keys(_bulkRotatePodsMap[appId].resources).forEach((kindName) => {
+                _bulkRotatePodsMap[appId].resources[kindName].isChecked = _bulkRotatePodsMap[appId].isChecked
+                _bulkRotatePodsMap[appId].resources[kindName].value = _bulkRotatePodsMap[appId].value
+                    ? CHECKBOX_VALUE.CHECKED
+                    : null
+            })
+        }
+        if (key === APP_DETAILS_TEXT.KIND_NAME && _bulkRotatePodsMap[appId].resources[_kindName]) {
+            // handling resource level value for checkbox
+            _bulkRotatePodsMap[appId].resources[_kindName].isChecked =
+                !_bulkRotatePodsMap[appId].resources[_kindName].isChecked
+            _bulkRotatePodsMap[appId].resources[_kindName].value = _bulkRotatePodsMap[appId].resources[_kindName]
+                .isChecked
+                ? CHECKBOX_VALUE.CHECKED
+                : null
+            // handling app level value for checkbox
+            // eslint-disable-next-line no-nested-ternary
+            _bulkRotatePodsMap[appId].value = Object.values(_bulkRotatePodsMap[appId].resources).every(
+                (_resource) => _resource.isChecked,
+            )
+                ? CHECKBOX_VALUE.CHECKED
+                : Object.values(_bulkRotatePodsMap[appId].resources).some((_resource) => _resource.isChecked)
+                  ? CHECKBOX_VALUE.INTERMEDIATE
+                  : null
+            _bulkRotatePodsMap[appId].isChecked =
+                _bulkRotatePodsMap[appId].value === CHECKBOX_VALUE.CHECKED ||
+                _bulkRotatePodsMap[appId].value === CHECKBOX_VALUE.INTERMEDIATE
+        }
+        setBulkRotatePodsMap(_bulkRotatePodsMap)
+        setAllAppsCheckboxValue(_bulkRotatePodsMap)
```

Comment:
Why are we setting same value in 2 states?

Reply:
2nd one is not a state setter its just a function call. Will rename it